projects/dsp-ui/src/lib/search/expert-search/expert-search.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, EventEmitter, Inject, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { KnoraApiConfig } from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken } from '../../core/core.module';
+import { AdvancedSearchParams, AdvancedSearchParamsService } from '../services/advanced-search-params.service';
+
+@Component({
+    selector: 'dsp-expert-search',
+    templateUrl: './expert-search.component.html',
+    styleUrls: ['./expert-search.component.scss']
+})
+export class ExpertSearchComponent implements OnInit {
+
+    /**
+     * @param gravsearchQuery Send the gravsearch query back.
+     */
+    @Output() gravsearchQuery = new EventEmitter<string>();
+
+    /**
+     * @param toggleExpertSearchForm Trigger toggle for extended search form.
+     */
+    @Output() toggleExpertSearchForm = new EventEmitter<boolean>();
+
+    expertSearchForm: FormGroup;
+
+    constructor(
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig,
+        private _searchParamsService: AdvancedSearchParamsService,
+        private fb: FormBuilder,
+        private _route: ActivatedRoute,
+        private _router: Router
+    ) { }
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    /**
+     * @ignore
+     * Initiate the form with predefined Gravsearch query as example.
+     */
+    private initForm() {
+        this.expertSearchForm = this.fb.group({
+            gravquery: [
+`PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+PREFIX incunabula: <${this.dspApiConfig.apiUrl}/ontology/0803/incunabula/simple/v2#>
+
+CONSTRUCT {
+    ?book knora-api:isMainResource true .
+    ?book incunabula:title ?title .
+
+} WHERE {
+    ?book a incunabula:book .
+    ?book incunabula:title ?title .
+}
+`,
+                Validators.required
```

Comment:
There should be a validator that checks for a user-provided offset.

Reply:
done in 79f3143 + bbc3af4

Reply:
ok